src/Extensions/Logging/Internal/OmexLogsEventSource.cs

```diff
@@ -0,0 +1,256 @@
+		{
+			m_machineInformation = machineInformation;
+			m_serviceContext = context;
+		}
+
+
+		private readonly IMachineInformation m_machineInformation;
+		private readonly IServiceContext m_serviceContext;
+
+		private string ApplicationName => m_machineInformation.MachineRole;
+		private string ServiceName => m_machineInformation.ServiceName;
+		private string BuildVersion => m_machineInformation.BuildVersion;
+		private string AgentName => string.Empty;
+
+		private Guid PartitionId => m_serviceContext.PartitionId;
+		private long ReplicaId => m_serviceContext.ReplicaOrInstanceId;
+
+		private string CorrelationId => Activity.Current?.Id ?? string.Empty;
+		private string TransactionId => (Activity.Current?.TraceId ?? default).ToHexString();
+
+
+		[NonEvent]
+		public void ServiceMessage(string message, LogLevel level, string category, EventId eventId, int threadId)
+		{
+			if (!IsEnabled())
+			{
+				return;
+			}
+
+			string tagName = eventId.Name;
+			string tagId = TagIdAsString(eventId.Id);
+
+			//Event methods should have all information as parameters so we are passing them each time
+			// Posible Breaking changes:
+			// 1. CorrelationId type changed from Guid ?? Guid.Empty
+			// 2. TransactionId type Changed from uint ?? 0u
+			// 3. ThreadId type Changed from string
+			// 4. TagName to events so it should be also send
+			switch (level)
+			{
+				case LogLevel.Trace:
+					LogSpamServiceMessage(ApplicationName, ServiceName, AgentName, BuildVersion, ProcessName, PartitionId, ReplicaId, CorrelationId, TransactionId, "Spam", category, tagId, tagName, threadId, message);
+					break;
+				case LogLevel.Debug:
+					LogVerboseServiceMessage(ApplicationName, ServiceName, AgentName, BuildVersion, ProcessName, PartitionId, ReplicaId, CorrelationId, TransactionId, "Verbose", category, tagId, tagName, threadId, message);
+					break;
+				case LogLevel.Information:
+					LogInfoServiceMessage(ApplicationName, ServiceName, AgentName, BuildVersion, ProcessName, PartitionId, ReplicaId, CorrelationId, TransactionId, "Info", category, tagId, tagName, threadId, message);
+					break;
+				case LogLevel.Warning:
+					LogWarningServiceMessage(ApplicationName, ServiceName, AgentName, BuildVersion, ProcessName, PartitionId, ReplicaId, CorrelationId, TransactionId, "Warning", category, tagId, tagName, threadId, message);
+					break;
+				case LogLevel.Error:
+				case LogLevel.Critical:
+					LogErrorServiceMessage(ApplicationName, ServiceName, AgentName, BuildVersion, ProcessName, PartitionId, ReplicaId, CorrelationId, TransactionId, "Error", category, tagId, eventId.Name, threadId, message);
+					break;
+				case LogLevel.None:
+					break;
+				default:
+					throw new ArgumentException("Unknown EventLevel: " + level);
+			}
+		}
+
+
+		[Event((int)EventIds.LogErrorEventId, Level = EventLevel.Error, Message = "{13}", Version = 6)]
+		private void LogErrorServiceMessage(
```

Comment:
>LogErrorServiceMessage [](start = 15, length = 22)

Including all of these parameters is a code smell. It's much more preferable to create some sort of property bag class which includes these parameters being passed around. And then you crack open the property bag at the last step of the process.

Reply:
This is an event source requirement, it can accept only simple types and parameter names used for event mapping